test/e2e_node/log_path_test.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_node
+
+import (
+	"bufio"
+	"os"
+	"strings"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util/uuid"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = framework.KubeDescribe("ContainerLogPath", func() {
+	f := framework.NewDefaultFramework("kubelet-container-log-path")
+	Describe("Pod containers", func() {
+		Context("with log printed to stdout", func() {
+			var ns, logDir string
+			var podClient *framework.PodClient
+			BeforeEach(func() {
+				podClient = f.PodClient()
+				ns = f.Namespace.Name
+				// logDir is the same with kubelet.containerLogsDir
+				logDir = "/var/log/containers/"
+			})
+			// NOTE: We can not do this check within a pod because container will try to find the source file of a symlink.
+			It("should print log to correct log path", func() {
+				podName := "log-pod" + string(uuid.NewUUID())
+				logContName := "log-container" + string(uuid.NewUUID())
+
+				podObject := &api.Pod{
+					ObjectMeta: api.ObjectMeta{
+						Name:      podName,
+						Namespace: ns,
+					},
+					Spec: api.PodSpec{
+						RestartPolicy: api.RestartPolicyNever,
+						Containers: []api.Container{
+							{
+								Image:   "gcr.io/google_containers/busybox:1.24",
+								Name:    logContName,
+								Command: []string{"sh", "-c", "for var in 1 2 3 4; do echo \"$(hostname)\"; sleep 1; done"},
+							},
+						},
+					},
+				}
+
+				podClient.Create(podObject)
+				err := framework.WaitForPodSuccessInNamespace(f.Client, podName, ns)
```

Comment:
Combined your two comments together, since we need to validate the contents, so after output logs, I will leave the container sleep. Do you think this is ok?